### PR TITLE
Provide filament (un)load user abort

### PR DESCRIPTION
**Pull request by espr14**

### CODE REVIEW

File: src/common/marlin_server.hpp [hunk 1]
```diff
@@ -25,6 +25,9 @@ void fsm_change(ClientFSM type, T phase, fsm::PhaseData data) {
     _fsm_change(type, fsm::BaseData(GetPhaseIndex(phase), data));
 }
 
+// true if user can stop waiting for heating/cooling by pressing a button
+extern bool can_stop_wait_for_heatup;
```
Comment (RadekVana) [inline src/common/marlin_server.hpp:29]: don't extern variables
include header containing getter/setter function declaration

btw it seams to be unused, is it dead code?

File: src/marlin_stubs/pause/M70X.cpp [hunk 2]
```diff
@@ -238,22 +239,26 @@ static PreheatStatus::Result M1400_no_parser(uint32_t val) {
     case PreheatMode::None:
         return Filaments::CurrentIndex() == filament_t::NONE ? PreheatStatus::Result::DoneNoFilament : PreheatStatus::Result::DoneHasFilament;
     case PreheatMode::Purge:
+        Pause::Instance().StopReset();
```
Comment (RadekVana) [inline src/marlin_stubs/pause/M70X.cpp:242]: note:
I think table (mask) containing which state should call `Pause::Instance().StopReset();` would be better

Comment (RadekVana) [reply src/marlin_stubs/pause/M70X.cpp:242]: or just do use one more switch before this one
```
switch (data.Mode()) {
    case PreheatMode:: some_state :
    ...
    case PreheatMode:: some_state :
      Pause::Instance().StopReset();
    default:
      break;
}
```

File: src/marlin_stubs/pause/pause.cpp [hunk 3]
```diff
@@ -263,12 +263,20 @@ bool Pause::ensureSafeTemperatureNotifyProgress(uint8_t progress_min, uint8_t pr
         return true;
     }
 
-    setPhase(PhasesLoadUnload::WaitingTemp, progress_min);
+    setPhase(can_stop ? PhasesLoadUnload::WaitingTemp_stoppable : PhasesLoadUnload::WaitingTemp_unstoppable, progress_min);
```
Comment (RadekVana) [inline src/marlin_stubs/pause/pause.cpp:266]: note:
this `can_stop ? ... : ...` is OK now
but we really should add transition table (and unify all state machines in FW)
it is my code, so it is not your fault 😃

**Review by RadekVana — CHANGES_REQUESTED** [review]

File: src/common/marlin_server.hpp (hunk 1 above)
Comment (espr14) [reply src/common/marlin_server.hpp:29]: A lot of added code with no advantage.

**Review by RadekVana — APPROVED** [review]

**Pull request merged**